generic3g/actions/MeanAction.F90

```diff
@@ -70,78 +83,63 @@ subroutine calculate_mean(this, rc)
 
    end subroutine calculate_mean
 
-   subroutine update_mean_accumulator(this, importState, exportState, clock, rc)
+   subroutine update_result_mean(this, rc)
       class(MeanAction), intent(inout) :: this
-      type(ESMF_State) :: importState
-      type(ESMF_State) :: exportState
-      type(ESMF_Clock) :: clock
       integer, optional, intent(out) :: rc
 
       integer :: status
-      
-      _ASSERT(this%initialized(), 'Accumulator has not been initialized.')
-      if(.not. this%update_calculated) then
-         call this%calculate_mean(_RC)
-      end if
-      call this%AccumulatorAction%update(importState, exportState, clock, _RC)
-      _RETURN(_SUCCESS)
-
-   end subroutine update_mean_accumulator
-
-   subroutine invalidate_mean_accumulator(this, importState, exportState, clock, rc)
-      class(MeanAction), intent(inout) :: this
-      type(ESMF_State) :: importState
-      type(ESMF_State) :: exportState
-      type(ESMF_Clock) :: clock
-      integer, optional, intent(out) :: rc
 
-      integer :: status
-      
-      call this%AccumulatorAction%invalidate(importState, exportState, clock, _RC)
-      this%counter_scalar = this%counter_scalar + 1
+      call this%calculate_mean(_RC)
+      call this%AccumulatorAction%update_result(_RC)
       _RETURN(_SUCCESS)
 
-   end subroutine invalidate_mean_accumulator
+   end subroutine update_result_mean
 
    subroutine calculate_mean_R4(this, rc)
       class(MeanAction), intent(inout) :: this
       integer, optional, intent(out) :: rc
 
       integer :: status
-      real(kind=ESMF_KIND_R4), pointer :: current_ptr(:) => null()
+      real(kind=ESMF_KIND_R4), pointer :: current_ptr(:)
+      integer(kind=COUNTER_KIND), pointer :: counter(:)
```

Comment:
I like `COUNTER_KIND`.   In hindsight it is unfortunate that ESMF did not do `ESMF_R4_KIND` instead of `...KIND_R4`.

Reply:
Agreed.